Automattic/vip-go-mu-plugins
# Change cache key in wpcom_vip_get_page_by_path()

## Description

The current cache key here is more or less unbounded, as a page url can be super long. This results in cache keys longer than memcached's 250 limit.

Worth noting that this function in core caches itself now: https://github.com/WordPress/WordPress/blob/1bf93a87a449a9476d1efd93a0452d7d6ceb3808/wp-includes/post.php#L5671-L5678. Though notably it does end up purged far more often than ours that has a persistent cache once a result is found.

This recent change (WP 6.3 milestone) looks to perhaps increase the time this cache would stay alive though: https://core.trac.wordpress.org/ticket/57625

Note that this deploy will result in a spike in queries for large sites relying on this cache. But the change warrants that IMO. But perhaps worth considering waiting and just relying on core WP's implementation eventually?

## Changelog Description
### Caching function updated

Changed the cache key & group for `wpcom_vip_get_page_by_path()` to avoid unbuo

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Call `wpcom_vip_get_page_by_path()` and make sure it still returns valid results.
